CoreSchemaView/src/au/gov/asd/tac/constellation/views/schemaview/providers/TransactionTypeNodeProvider.java

```diff
@@ -180,54 +180,25 @@ private void populateTree() {
     }
 
     private boolean isFilterMatchCurrentNode(SchemaTransactionType treeItem) {
-        boolean found = isFilterMatchText(treeItem.getName());
-        if (!found) {
-            found = isFilterMatchAnyProperty(treeItem);
-        }
-        return found;
+        return filterText.getText().toLowerCase().isEmpty() ? false
+                : isFilterMatchText(treeItem.getName()) || isFilterMatchAnyProperty(treeItem);
```

Comment:
@antares1470 I added original line to avoid further processing when the string is null. Feels like the above will still try to process when it is null? (I yet have to remove the .toLowerCase() from the check)

Reply:
@Auriga2 So in this case (and some of the others), Java will lazily evaluate the or boolean conditions (i.e. if the first condition is true it doesn't evaluate the rest of them). 

If you want to avoid further processing for null, `isEmpty()` won't actually check that (in fact you'll get null pointer calling a `isEmpty()` on null). If you want to include a null check, change `!filterText.getText().isEmpty()` to `StringUtils.isNotBlank(filterText.getText())`.

Ultimately, the purpose of my suggestions is to remove the literal boolean false from the statement

Reply:
Yeah when it (either) is empty, that side is false, so it valuates the other side because of False ||.
Looks like we should add an && there.
Sorry I meant empty. But checking for both could be better.

Reply:
My bad, you are correct (I was mixing up m translations). It should be `!filterText.getText().isEmpty() && (isFilterMatchText(treeItem.getName()) || isFilterMatchAnyProperty(treeItem))`

I'll edit the other comments accordingly

Reply:
I think it's safer to use isNotBlank. I edited them all, and will test a bit and push in a while.